File: peek_plugin_diagram/_private/tuples/DiagramLoaderStatusTuple.py
from peek_plugin_diagram._private.PluginNames import diagramTuplePrefix
from vortex.Tuple import addTupleType, TupleField, Tuple


@addTupleType
class DiagramLoaderStatusTuple(Tuple):
    __tupleType__ = diagramTuplePrefix + "DiagramLoaderStatusTuple"

    displayCompilerQueueStatus: bool = TupleField(False)
    displayCompilerQueueSize: int = TupleField(0)
    displayCompilerProcessedTotal: int = TupleField(0)
    displayCompilerLastError: str = TupleField()

    gridCompilerQueueStatus: bool = TupleField(False)
    gridCompilerQueueSize: int = TupleField(0)
    gridCompilerQueueProcessedTotal: int = TupleField(0)
    gridCompilerQueueLastError: str = TupleField()

File: peek_plugin_diagram/plugin-module/_private/tuples/DiagramLoaderStatusTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {diagramTuplePrefix} from "../PluginNames";


@addTupleType
export class DiagramLoaderStatusTuple extends Tuple {
    public static readonly tupleName = diagramTuplePrefix + "DiagramLoaderStatusTuple";

    displayCompilerQueueStatus: boolean;
    displayCompilerQueueSize: number;
    displayCompilerProcessedTotal: number;
    displayCompilerLastError: string;

    gridCompilerQueueStatus: boolean;
    gridCompilerQueueSize: number;
    gridCompilerQueueProcessedTotal: number;
    gridCompilerQueueLastError: string;

    constructor() {
        super(DiagramLoaderStatusTuple.tupleName)
    }
}